contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -145,4 +157,230 @@ contract OETHVaultCore is VaultCore {
+     * @return amount Amount of WETH transferred to the withdrawer
+     */
+    function claimWithdrawal(uint256 _requestId)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256 amount)
+    {
+        amount = _claimWithdrawal(_requestId);
+
+        // transfer WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, amount);
+    }
+
+    /**
+     * @notice Claim a previously requested withdrawals once they are claimable.
+     * This requests can be claimed once the withdrawal queue's `claimable` amount
+     * is greater than or equal each request's `queued` amount.
+     * If one of the requests is not claimable, the whole transaction will revert with `queue pending liquidity`.
+     * @param _requestIds Unique ID of each withdrawal request
+     * @return amounts Amount of WETH received for each request
+     * @return totalAmount Total amount of WETH transferred to the withdrawer
+     */
+    function claimWithdrawals(uint256[] memory _requestIds)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256[] memory amounts, uint256 totalAmount)
+    {
+        amounts = new uint256[](_requestIds.length);
+        for (uint256 i = 0; i < _requestIds.length; ++i) {
+            amounts[i] = _claimWithdrawal(_requestIds[i]);
+            totalAmount += amounts[i];
+        }
+
+        // transfer all the claimed WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, totalAmount);
+    }
+
+    // slither-disable-start reentrancy-no-eth
+
+    function _claimWithdrawal(uint256 requestId)
+        internal
+        returns (uint256 amount)
+    {
+        // Check if there's enough liquidity to cover the withdrawal request
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+        WithdrawalRequest memory request = withdrawalRequests[requestId];
+
+        require(request.claimed == false, "Already claimed");
+        require(request.withdrawer == msg.sender, "Not requester");
+        require(
+            request.timestamp + CLAIM_DELAY <= block.timestamp,
+            "Claim delay not met"
+        );
+
+        // Try and get more liquidity in the withdrawal queue if there is not enough
+        if (request.queued > queue.claimable) {
+            // Stream any harvested rewards (WETH) that are available to the Vault
+            IDripper(dripper).collect();
+
+            // Add any WETH from the Dripper to the withdrawal queue
+            uint256 addedClaimable = _addWithdrawalQueueLiquidity();
+
+            // If there still isn't enough liquidity in the queue to claim, revert
+            require(
+                request.queued <= queue.claimable + addedClaimable,
+                "Queue pending liquidity"
+            );
+        }
+
+        // Store the updated claimed amount
+        withdrawalQueueMetadata.claimed = queue.claimed + request.amount;
```

Comment:
nit: Any reason for not doing a `queue.claimable -= request.amount` here? I feel like that'll simplify the calculations in the other places like `wethAvailable`

Reply:
you mean `queue.claimable += request.amount`?

`queue` is an in memory variable. We need to update the storage variable `withdrawalQueueMetadata`.
`withdrawalQueueMetadata.claimed += request.amount` would work but that would do an extra read from storage. So I've done with adding the two amounts using in memory variables and then storing back to storage.

Reply:
Nope, I meant that we can subtract claimed amount from `claimable` (I see that we do a `queue.claimable - queue.claimed` in a lot of places that could be avoided. Besides `claimable` still including claimed amount seems a bit off since I am thinking `claimable` == `unclaimed`)

Reply:
`queue.claimable` is important as it drives when a request can be claimed. It's a global variable that goes up whenever liquidity is reserved for the queue. A request can be claimed when this invariant is true `request.queued <= queue.claimable`. 

`claimable != unclaimed` as `uint256 unclaimed = queue.claimable - queue.claimed;`
`queue.claimed` is also a global storage variable of all requests that have every been claimed.

The advantage of using the global `queued` and `claimable` is we don't need to iterate over user requests to work out if the request can be claimed or not.